packages/near-membrane-base/src/membrane.ts

```diff
@@ -611,6 +644,34 @@ export function createMembraneMarshall() {
             return fn;
         }
 
+        function isLiveObject(value: any): boolean {
+            if (value === null || value === undefined || value === ObjectProto) {
+                return false;
+            }
+            if (typeof value === 'object') {
+                if (
+                    // Arrays are marked as live in the BoundaryProxyHandler constructor.
+                    // isArrayOrNotOrThrowForRevoked(value) ||
+                    ArrayBufferIsView(value)
+                ) {
+                    return true;
+                }
+                const { constructor } = value;
+                if (constructor === ObjectCtor) {
+                    // If the constructor, own or inherited, points to `Object`
+                    // then `value` is not likely a prototype object.
+                    return true;
+                }
+                if (ReflectGetPrototypeOf(value) === null) {
+                    // Ensure `value` is not an `Object.prototype` from an iframe.
```

Comment:
why do we care about this? if it is the `Object.prototype` value from any realm it should never be mutate.

Reply:
`Object.prototype` has a `ReflectGetPrototypeOf(value) === null`. So if we don't guard for it then it'll pass and be mutable.